packages/openapi-v3/src/controller-spec.ts

```diff
@@ -178,16 +178,34 @@ function resolveControllerSpec(constructor: Function): ControllerSpec {
         if (!spec.components.schemas) {
           spec.components.schemas = {};
         }
+        if (p.name in spec.components.schemas) {
+          // Preserve user-provided definitions
```

Comment:
We don't have any test for this branch in `openapi-v3` testsuite. Should we add one?